src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/ProgramStageSelector/ProgramStageSelector.container.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+import React, { useMemo } from 'react';
+import i18n from '@dhis2/d2-i18n';
+import { useHistory } from 'react-router-dom';
+import log from 'loglevel';
+import { ProgramStageSelectorComponent } from './ProgramStageSelector.component';
+import { Widget } from '../../../Widget';
+import { errorCreator } from '../../../../../capture-core-utils';
+import { useCommonEnrollmentDomainData } from '../../common/EnrollmentOverviewDomain';
+import type { Props } from './ProgramStageSelector.types';
+import { useProgramFromIndexedDB } from '../../../../utils/cachedData/useProgramFromIndexedDB';
+import { useLocationQuery, buildUrlQueryString } from '../../../../utils/routing';
+
+
+export const ProgramStageSelector = ({ programId, orgUnitId, teiId, enrollmentId }: Props) => {
+    const history = useHistory();
+    const { tab } = useLocationQuery();
+    const { error: enrollmentsError, enrollment } = useCommonEnrollmentDomainData(teiId, enrollmentId, programId);
+    const {
+        loading: programLoading,
+        error: programError,
+        program,
+    } = useProgramFromIndexedDB(programId);
+
+
+    if (enrollmentsError || programError) {
+        log.error(errorCreator('Enrollment page could not be loaded')({
+            enrollmentsError,
+            programError,
+        }));
+    }
+
+    const programStages = useMemo(() => !programLoading && program?.programStages?.reduce((accStage, currentStage) => {
+        accStage.push({
+            id: currentStage.id,
+            eventCount: (enrollment?.events
+                ?.filter(event => event.programStage === currentStage.id)
+                ?.length
+            ),
+            displayName: currentStage.displayName,
+            style: currentStage.style,
+            repeatable: currentStage.repeatable,
+        });
+        return accStage;
+    }, []), [enrollment?.events, program?.programStages, programLoading]);
+
+    const onSelectProgramStage = (newStageId: string) =>
+        history.push(`enrollmentEventNew?${buildUrlQueryString({
```

Comment:
I changed this up a bit. Made a new function called `buildUrlQueryString` and used that instead of `urlArguments`. You did follow the established pattern here, so nothing wrong with what you did, but I found the `urlArguments` implementation to not be the best.

I also made a follow up issue to replace the usage of `urlArguments` with `buildUrlQueryString`: https://jira.dhis2.org/browse/TECH-794. This is not related to this PR at all, but if you find the time it would be great if you can refactor to use the new function instead of `urlArguments` everywhere in the app.

Reply:
Yep, I'll have a look at TECH-794 👍 Thanks!